src/libraries/Common/tests/System/Security/Cryptography/PlatformSupport.cs

```diff
@@ -89,7 +89,35 @@ private static bool CheckIfVbsAvailable()
         internal const TestPlatforms OpenSSL = TestPlatforms.AnyUnix & ~(AppleCrypto | TestPlatforms.Android | TestPlatforms.Browser);
 
         // Whether or not the current platform supports RC2
-        internal static readonly bool IsRC2Supported = !PlatformDetection.IsAndroid;
+        internal static bool IsRC2Supported
+        {
+            get
+            {
+                if (PlatformDetection.IsAndroid)
+                {
+                    return false;
+                }
+
+                if (PlatformDetection.IsLinux)
+                {
+                    try
+                    {
+                        using (RC2 rc2 = RC2.Create())
+                        using (rc2.CreateEncryptor())
+                        {
+                        }
+
+                        return true;
```

Comment:
This should be memoized.  Either via System.Lazy or just introducing a static to hold on to "do we already know the answer?"  e.g. 0 for unknown, 1 for known true and -1 for known false.